### DISCUSSION TITLE
Query with LIMIT inside JOIN does not work

### DISCUSSION THREAD

**Discussion by huytool157** [planetscale/discussion#393] [Q&A] [answered]
This query does not work with Planetscale

```
SELECT
  c.column_name as column_name,
  c.data_type as data_type,
  c.table_name as table_name,
  c.table_schema as table_schema
FROM
  information_schema.columns c
  JOIN (
    SELECT
      table_name
    FROM
      information_schema.tables
    WHERE
      table_schema != 'information_schema'
    LIMIT
      1
  ) AS tables ON tables.table_name = c.table_name
ORDER BY
  c.table_name
```

If you remove `LIMIT 1` then it works, apparently this is a [problem](https://stackoverflow.com/questions/12810346/alternative-to-using-limit-keyword-in-a-subquery-in-mysql) with older MySQL versions but I'm not sure if it applies here. Is there an alternative query for the above that I can use?

**Comment by orware** [comment]
@huytool157,

We've reported the issue internally to our Vitess team for further review. Thank you for sharing the query/issue with us!

**Reply by huytool157** [reply]
@orware hello, is there an update for this issue? This is affecting our customer's Planetscale integration.

**Reply by orware** [reply]
@huytool157,

The issue has been reported, but I do not currently have an update available to share for you yet.

**Reply by huytool157** [reply]
Hey @orware any updates you can share with us? Any chance we can escalate the issue because is pretty specific to Planetscale (this query works for other MySQL database integrations in our platform and we have lots of them).

**Reply by orware** [reply]
@huytool157,

No immediate updates aside from seeing that the internal issue I've created is on the todo list for Vitess v17. 

Vitess v16 is expected to be released next week, so that would provide an approximate time of about 4 months for Vitess v17 to be released since the Vitess project is on a 4-month release cadence.

**Reply by aaly00** [reply]
Does this mean that it'll take four months to fix this bug ?

**Reply by orware** [reply]
@aaly00, That could be a possibility, unless it gets fixed earlier and rolled into our PlanetScale Vitess images sooner than that. 

If that occurs I'll definitely provide an update here to let you know!

**Reply by huytool157** [reply]
Is there an alternative query we can use?

**Reply by aaly00** [reply]
May I suggest using this query instead. 

```SQL 
          select
                   col.column_name as column_name,
                   col.data_type as data_type,
                   col.table_name as table_name,
                   col.table_schema as table_schema

            from information_schema.tables as tab
                     inner join information_schema.columns as col
                                on col.table_schema = tab.table_schema
                                    and col.table_name = tab.table_name
            where tab.table_type = 'BASE TABLE'
              and tab.table_schema = database()
            order by tab.table_name,
                     col.ordinal_position;
```

**Reply by orware** [reply, marked as answer]
@huytool157,

I received an update from our Vitess team that once [PR#12634](https://github.com/vitessio/vitess/pull/12634) is merged in and eventually makes its way into PlanetScale in a few weeks more generally that this should address the issue reported here, even though it looks like a separate patch is also already available from the Retool team based on what I saw in an update in the related [Retool discussion](https://community.retool.com/t/select-table-shows-no-data-in-gui-mode-query/7816/27?u=orware).

**Reply by huytool157** [reply]
Thanks for the quick turnaround! We're prepared to update our codebase on our end.

**Reply by huytool157** [reply]
Hey @orware  has the fix made it to production PlanetScale?

**Reply by orware** [reply]
@huytool157,

Sorry I meant to circle back on this after your last ping back on April 20th about this.

I just ran the original query shared above and it does seem to be working now based on what I can see:
![image](https://user-images.githubusercontent.com/931650/235509818-23c87f5a-6020-4d4e-98af-7064d6851ed2.png)

**Reply by huytool157** [reply]
Looks good, do all Planetscale databases have this latest Vitess update?

**Reply by orware** [reply]
I would say at this point they should since it's been more than a few weeks since this was folded into the Vitess releases we include so it should be pretty much generally available at this time 👍 .

**Reply by huytool157** [reply]
@orware I'm back after six months. A few customers reported this issue again (we can repro). Did you guys make any changes to Vitess that might regress this query?

**Reply by orware** [reply]
@huytool157,

I performed some additional testing of the query shared above in a few test databases and it appears that the issue that had been fixed in the earlier pull request (https://github.com/vitessio/vitess/pull/12634) seems to be present in Vitess 17. Databases have been updating to that Vitess version over the last few weeks, including some more recent batches this week, which is likely why some users may be reporting encountering the issue again now.

Let me get this reported to our Vitess team and hopefully it won't take too long to follow-up with a solution but it may take a few days.

**Reply by JoeBumbaca** [reply]
@orware Just wanted to check in on this.  Is there any update?

**Reply by orware** [reply]
@JoeBumbaca,

A fix was put together by our Vitess team yesterday over in https://github.com/vitessio/vitess/pull/14501 so the fix should be making its way into our Vitess 17 builds fairly soon, possibly as early as next week but due to our staggered deployments there may be some additional delay so I'll try and provide an update here if I hear anything concrete about the fix being available.

For anyone urgently impacted by the issue I do have a temporary workaround available that I can apply to any branches being used with Retool so for any of those situations please write-in over at https://support.planetscale.com/ and we should be able to help you using that option if needed.

**Reply by orware** [reply]
Hi everyone 👋 ,

I was testing out our current Vitess versions since my last update a few weeks ago and it looks like the fix above has made its way into our current builds so any new/existing databases should be back to working well with Retool again from what I can see!
